seed/analysis_pipelines/bsyncr.py

```diff
@@ -252,3 +286,192 @@ def _build_bsyncr_input(analysis_property_view, meter):
+    analysis_property_view_id_elem = input_file_tree.xpath(id_xpath, namespaces=NAMESPACES)
+
+    if len(analysis_property_view_id_elem) != 1:
+        raise AnalysisPipelineException(f'Expected BuildlingSync file to have exactly one "{PREMISES_ID_NAME}" PremisesIdentifier')
+    return int(analysis_property_view_id_elem[0].text)
+
+
+@shared_task
+def _start_analysis(analysis_id, progress_data_key):
+    """Start bsyncr analysis by making requests to the service
+
+    """
+    analysis = Analysis.objects.get(id=analysis_id)
+    analysis.status = Analysis.RUNNING
+    analysis.save()
+
+    progress_data = ProgressData.from_key(progress_data_key)
+    progress_data.step('Sending requests to bsyncr service')
+
+    output_file_ids = []
+    for input_file in analysis.input_files.all():
+        try:
+            analysis_property_view_id = _parse_analysis_property_view_id(input_file.file.path)
+            result, errors = _run_bsyncr_analysis(input_file.file)
+            if errors:
+                for error in errors:
+                    AnalysisMessage.log_and_create(
+                        logger=logger,
+                        type_=AnalysisMessage.ERROR,
+                        analysis_id=analysis.id,
+                        analysis_property_view_id=analysis_property_view_id,
+                        user_message='Unexpected error from bsyncr service',
+                        debug_message=error,
+                    )
+                continue
+
+            analysis_output_file = AnalysisOutputFile(
+                content_type=AnalysisOutputFile.BUILDINGSYNC,
+            )
+            padded_id = f'{analysis_property_view_id:06d}'
+            analysis_output_file.file.save(f'bsyncr_output_{padded_id}.xml', ContentFile(result))
+            analysis_output_file.clean()
+            analysis_output_file.save()
+            analysis_output_file.analysis_property_views.set([analysis_property_view_id])
+            output_file_ids.append(analysis_output_file.id)
+        except Exception as e:
+            AnalysisMessage.log_and_create(
+                logger=logger,
+                type_=AnalysisMessage.ERROR,
+                analysis_id=analysis.id,
+                user_message='Unexpected error while running analysis',
+                debug_message='',
+                exception=e,
+            )
+            continue
+
+    if len(output_file_ids) == 0:
+        pipeline = BsyncrPipeline(analysis.id)
+        message = 'Failed to get results for all properties'
+        pipeline.fail(message)
+        # stop the task chain
+        raise AnalysisPipelineException(message)
+
+    return output_file_ids
+
+
+@shared_task
+def _process_results(analysis_output_file_ids, analysis_id, progress_data_key):
+    analysis = Analysis.objects.get(id=analysis_id)
+    analysis.status = Analysis.RUNNING
+    analysis.save()
```

Comment:
Might be remnant from copying the first task, or are there cases where the status wouldn't already be `RUNNING`? Even if this were redundant, it's non-breaking so not blocking merge.

Reply:
Whoops, yes this is a non-critical bug. It shouldn't be trying to update the status, I'll remember to remove this later